### DISCUSSION TITLE
Ability to change the path of api using next.config or env

### DISCUSSION THREAD

**Discussion by afzaalahmad** [vercel/next.js#35229] [Ideas]
### Describe the feature you'd like to request

Currently the path of api is `/api` and it is hardcoded regex in constants file. I am suggesting a change to make it feasible for a user to change the api path to any other using either `next.config` or `env`. `next.config` should have an option to set custom path for the api. We can use `apiPath` key in `next.config` similar to `basePath` in order to have a customised path e.g. if I want to change the path from `/api` to `/raw`.

### Describe the solution you'd like

We can simply load API_ROUTE from `next.config` key `apiPath` as a regex or string, instead of hardcoding.

### Describe alternatives you've considered

None.

**Comment by arodik** [comment]
I agree. We host our own backend API under `/api,` so I can't use API routes at all. Would be great if we had an ability to change API routes path to something else
